### ISSUE TITLE
doesn't build on 1.1 release

### ISSUE THREAD

**Issue by blaenk** [cyderize/rust-websocket#44]
TL;DR: Quick (and IMO correct) fix for now is to pin the hyper dependency to the last version before the refactor, 0.5.2. Iron [did the same thing](https://github.com/iron/iron/blob/master/Cargo.toml#L23) while it gets familiar with the changes.

So 1.1 is out. Hyper made some refactorings that changed some stuff around, but mainly with respect to this package it concerns the fact that `parse_response` is now in `hyper::http::h1` instead of just `hyper::http`, and so on. I was working on a fix but the problem I ran into was this at the end of `src/server/mod.rs`:

``` rust
/// Represents a connection to the server that has not been processed yet.
pub struct Connection<R: Read, W: Write>(R, W);

impl<R: Read, W: Write> Connection<R, W> {
	/// Process this connection and read the request.
	pub fn read_request(self) -> io::Result<Request<R, W>> {
		match Request::read(self.0, self.1) {
			Ok(result) => { Ok(result) },
			Err(err) => {
				Err(io::Error::new(io::ErrorKind::InvalidInput, err))
			}
		}
	}
}
```

This `io::Error::new` method requires that the error payload `err` 'implements' `Into<Box<Error + Send + Sync>>`. The error is that the `err` is not `Sync` or `Send`. These kinds of bugs are annoying to track down because it's difficult to tell which field is implicitly breaking the `Sync/Send` bounds. If I had to guess, it's also coming from the hyper refactors, so one of the hyper-related error variants is most likely the culprit.

**Comment by jamii** [comment]
https://github.com/mlalic/solicit/issues/6

**Comment by blaenk** [comment]
Awesome find.

**Comment by jamii** [comment]
I hacked around it in https://github.com/jamii/rust-websocket for now while we wait for the upstream fix.

**Comment by jamii** [comment]
The underlying error is fixed - https://github.com/mlalic/solicit/issues/6#issuecomment-115399490

**Comment by durka** [comment]
This is still an issue and makes the "build: passing" sticker in the README a lie :(

For my own project I had to override the websocket dependency and fix Cargo.toml.

**Comment by cyderize** [comment]
Fixed at version 0.12.2
